internal/confgen/document_parser.go

```diff
@@ -527,13 +527,37 @@ func (sp *documentParser) parseUnionMessage(field *Field, msg protoreflect.Messa
 					)
 					return xerrors.WrapKV(xerrors.E2014(subField.opts.Name), kvs...)
 				}
-				err = sp.parser.parseUnionMessageField(subField, msg, valNode.Value)
+				var nodeDatas []string

```

Comment:
naming: `nodeDatas` -> `nodeDataList` or `nodeValues`

NOTE: `data` is an uncountable noun, so there is no `datas` plural word!